From 5a78b6d12739df5e551a56f6c93b6a5d4a791b18 Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Sat, 24 Feb 2024 22:43:57 -0800
Subject: [PATCH] don't mess with qsort_r

The built-in qsort_r implementation detection/test thing is broken on
Alpine, this makes it seem like fixing this upstream isn't easy:

https://github.com/rlancaste/stellarsolver/issues/125
---
 CMakeLists.txt | 26 +++-----------------------
 1 file changed, 3 insertions(+), 23 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ce09934..bddf9a2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,29 +58,9 @@ if (WIN32)
 
 else(WIN32)
 
-try_run(RUN_RESULT_2 COMPILE_SUCCESS_2 ${CMAKE_CURRENT_BINARY_DIR} ${config_SRCS} COMPILE_DEFINITIONS -DTEST_DECLARE_QSORT_R=ON)
-if(COMPILE_SUCCESS_2 AND (RUN_RESULT_2 EQUAL 0))
-    SET(VAR_2 0)
-else()
-    SET(VAR_2 1)
-endif()
-file(APPEND "${config_FN}" "#define NEED_DECLARE_QSORT_R ${VAR_2}\n")
-
-try_run(RUN_RESULT_3 COMPILE_SUCCESS_3 ${CMAKE_CURRENT_BINARY_DIR} ${config_SRCS} COMPILE_DEFINITIONS -DTEST_QSORT_R=ON)
-if(COMPILE_SUCCESS_3 AND (RUN_RESULT_3 EQUAL 0))
-    SET(VAR_3 1)
-else()
-    SET(VAR_3 0)
-endif()
-file(APPEND "${config_FN}" "#define NEED_QSORT_R ${VAR_3}\n")
-
-try_run(RUN_RESULT_4 COMPILE_SUCCESS_4 ${CMAKE_CURRENT_BINARY_DIR} ${config_SRCS} COMPILE_DEFINITIONS -DTEST_SWAP_QSORT_R=ON)
-if(COMPILE_SUCCESS_4 AND (RUN_RESULT_4 EQUAL 0))
-    SET(VAR_4 1)
-else()
-    SET(VAR_4 0)
-endif()
-file(APPEND "${config_FN}" "#define NEED_SWAP_QSORT_R ${VAR_4}\n")
+file(APPEND "${config_FN}" "#define NEED_DECLARE_QSORT_R 0\n")
+file(APPEND "${config_FN}" "#define NEED_QSORT_R 0\n")
+file(APPEND "${config_FN}" "#define NEED_SWAP_QSORT_R 1\n")
 
 endif(WIN32)
 
-- 
2.43.2

